### PR TITLE
Remove version constraint for AllenNLP and run `allennlp_*.py` on GitHub Actions

### PR DESCRIPTION
**Pull request by himkt**
This PR removes a version constraint for AllenNLP.
It also changes `examples.yml` such that GitHub Actions run `allennlp_*.py`.

## Motivation
Remove a specific version constraint.

## Description of the changes
AllenNLP supports Python3.8 since 1.2.0rc1 (https://github.com/allenai/allennlp/releases/tag/v1.2.0rc1).
And now we can use Optuna with the latest AllenNLP so I remove a version constraint.

### CODE REVIEW

File: tests/integration_tests/allennlp_tests/tiny_single_id.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 from allennlp.data.token_indexers.token_indexer import IndexedTokenList
 from allennlp.data.token_indexers.token_indexer import TokenIndexer
-from allennlp.data.tokenizers.token import Token
+from allennlp.data.tokenizers.token_class import Token
```
Comment (himkt) [inline tests/integration_tests/allennlp_tests/tiny_single_id.py:8]: `allennlp.data.tokenizers.token` was renamed to `allennlp.tokenizers.token_class` in v1.3.0.
(https://github.com/allenai/allennlp/releases/tag/v1.3.0)

I simply replace `token` with `token_class`, as `tiny_single_id.py` is used only in tests.

**Comment by himkt** [comment]
I confirmed AllenNLP examples works well in the local PR.
(details: https://github.com/himkt/optuna/actions/runs/493613329)

**Comment by toshihikoyanase** [comment]
@keisuke-umezawa Could you review this PR, please? I think the changes can be simply examined with the tests and examples.

File: setup.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def get_extras_require() -> Dict[str, List[str]]:
             "torchvision==0.8.2 ; sys_platform=='darwin'",
             "torchvision==0.8.2+cpu ; sys_platform!='darwin'",
             "torchaudio==0.7.2",
-            "allennlp==1.2.0 ; python_version<'3.8'",
+            "allennlp ; python_version<'3.8'",
```
Comment (toshihikoyanase) [inline setup.py:185]: [Question] This is out of the scope of this PR, do we have still problems to execute the AllenNLP related tests in Python 3.8?

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your PR. I confirmed that the examples worked in my environment (Ubuntu 20.04, Python 3.8.3). LGTM!

File: setup.py (hunk 2 above)
Comment (himkt) [reply setup.py]: Thank you for the catch!
d5d5184 removes version constraint in integration section.

Comment (toshihikoyanase) [reply setup.py]: Thank! LGTM, again.

**Review by keisuke-umezawa — APPROVED** [review]
LGTM!

**Pull request merged**